src/Hyperlink.js

```diff
@@ -77,7 +77,7 @@ class Hyperlink extends Component {
               ? this.props.linkText(url)
               : this.props.linkText
 
-        if (OS !== 'web') {
+        if (OS !== 'web' && this.props.onLongPress) {
           componentProps.onLongPress = () => this.props.onLongPress && this.props.onLongPress(url, text)
```

Comment:
`componentProps` is reused on line 95. This means `onLongPress` goes over the entire text and not just the link portion of the text